gimme_aws_creds/okta.py

```diff
@@ -37,6 +41,50 @@
 from .registered_authenticators import RegisteredAuthenticators
 
 
+class OktaCookieJar(RequestsCookieJar):
+    """Extends RequestsCookieJar adding serialization to/from a file."""
+    def __init__(self, path, *args, **kwargs):
+        """Populates the jar from filename if the file exists."""
+        super().__init__(*args, **kwargs)
+        self._path = path
+        # Use a temporary LWPCookieJar to read the cookies from disk
+        jar = LWPCookieJar(path.as_posix())
+        try:
+            jar.load(ignore_discard=True)
+        except FileNotFoundError:
+            pass  # jar may not exist on first run
+        for cookie in jar:
+            super().set_cookie(cookie)
+
+    def save(self):
+        """Mimics LWPCookieJar.save but with multi-process safety."""
```

Comment:
Code here: https://github.com/python/cpython/blob/main/Lib/http/cookiejar.py#L1892-L1902

Reply:
Just a suggestion, I wonder if you can instead create a `jar = LWPCookieJar()` and copy the cookies:

```
jar = LWPCookieJar()
for cookie in self:
  if cookie.is_expired(now):
     continue
  jar.set_cookie(cookie)
jar.save(tmp.name)
```

Although that opens the file again and the code is rather simple